situp-plugins/service-map-stateful/src/main/java/com/amazon/situp/plugins/processor/ServiceMapStatefulProcessor.java

```diff
@@ -38,34 +38,30 @@
     private volatile static LmdbProcessorState<String> currentTraceGroupWindow;
     //TODO: Consider keeping this state in lmdb
     private volatile static  HashSet<ServiceMapRelationship> relationshipState = new HashSet<>();
-    private static File databasePath;
-    private static File traceDatabasePath;
+    private static String dbPath;
     private static Clock clock;
 
     private final int thisProcessorId;
 
     public ServiceMapStatefulProcessor(final PluginSetting pluginSetting) {
      this(pluginSetting.getIntegerOrDefault(ServiceMapProcessorConfig.WINDOW_DURATION, ServiceMapProcessorConfig.DEFAULT_WINDOW_DURATION)*TO_MILLIS,
              new File(ServiceMapProcessorConfig.DEFAULT_SPANS_LMDB_PATH),
-             new File(ServiceMapProcessorConfig.DEFAULT_TRACES_LMDB_PATH),
              Clock.systemUTC());
     }
 
     ServiceMapStatefulProcessor(final long windowDurationMillis,
                                        final File databasePath,
-                                       final File traceDatabasePath,
                                        final Clock clock) {
         ServiceMapStatefulProcessor.clock = clock;
         this.thisProcessorId = processorsCreated.getAndIncrement();
         if(isMasterInstance()) {
             previousTimestamp = ServiceMapStatefulProcessor.clock.millis();
             ServiceMapStatefulProcessor.windowDurationMillis = windowDurationMillis;
-            ServiceMapStatefulProcessor.databasePath = createPath(databasePath);
-            ServiceMapStatefulProcessor.traceDatabasePath = createPath(traceDatabasePath);
-            currentWindow = new LmdbProcessorState<>(databasePath, getNewDbName(), ServiceMapStateData.class);
-            previousWindow = new LmdbProcessorState<>(databasePath, getNewDbName() + "-prev", ServiceMapStateData.class);
-            currentTraceGroupWindow = new LmdbProcessorState<>(traceDatabasePath, getNewTraceDbName(), String.class);
-            previousTraceGroupWindow = new LmdbProcessorState<>(traceDatabasePath, getNewTraceDbName() + "-prev", String.class);
+            ServiceMapStatefulProcessor.dbPath = databasePath.getPath();
```

Comment:
I guess this will not work, MDB_NOSUBDIR doesn't create the path to file. The reason testcases work is because Temporaryfolder.newfile() is valid. Add a test case with default config to confirm this works?

Note: The test case should create resource in the default config and delete it.

Reply:
So im not sure why MDB_NOSUBDIR would need to create the path to the file. Without the flag it doesnt create the path either. Not really sure what the issue is here (I guess I can use that createPath function to be safe), but I can make the test case change.

Reply:
The default path is `/data/service-ma/*`. If the directory doesn't exists the plugin crashes. 


`
Caused by: org.lmdbjava.LmdbNativeException$ConstantDerivedException: Platform constant error code: ENOENT No such file or directory (2)
	at org.lmdbjava.ResultCodeMapper.checkRc(ResultCodeMapper.java:114)
	at org.lmdbjava.Env$Builder.open(Env.java:495)
	at org.lmdbjava.Env$Builder.open(Env.java:512)
	at com.amazon.situp.plugins.processor.state.LmdbProcessorState.<init>(LmdbProcessorState.java:41)
	at com.amazon.situp.plugins.processor.ServiceMapStatefulProcessor.<init>(ServiceMapStatefulProcessor.java:61)
	at com.amazon.situp.plugins.processor.ServiceMapStatefulProcessor.<init>(ServiceMapStatefulProcessor.java:47)
	... 19 more
`

Fix,

`
 ServiceMapStatefulProcessor.dbPath = createPath(databasePath).getPath();
`